src/Stack/Config.hs

```diff
@@ -324,7 +323,20 @@ loadBuildConfig mproject config stackRoot mresolver = do
                            , projectFlags = mempty
                            , projectResolver = r
                            }
-                   liftIO $ Yaml.encodeFile dest' p
+                   liftIO $ do
+                       S.writeFile dest'
+                           ("# This is the implicit global project's config file, which is only used when\n" <>
```

Comment:
This bit of code has bad complexity characteristics. It would be better to use `S.concat` or a `Builder`.